api/hypershift/v1beta1/nodepool_types.go

```diff
@@ -997,6 +1026,39 @@ type AzureNodePoolPlatform struct {
 	MachineIdentityID string `json:"machineIdentityID,omitempty"`
 }
 
+// MarketplaceImage specifies the information needed to create an Azure VM from an Azure Marketplace image. This struct
+// replicates the same fields found in CAPZ - https://github.com/kubernetes-sigs/cluster-api-provider-azure/blob/main/api/v1beta1/types.go.
+type MarketplaceImage struct {
```

Comment:
why does this need to be API facing? wouldn't all this be known information owned by us that would make it an impl detail?

Reply:
`AzureMachineTemplate` needs these fields for an Azure Marketplace image ([ref here](https://capz.sigs.k8s.io/topics/custom-images.html#using-azure-marketplace)).

Reply:
I think is fine to expose that in the API to enable compositional use cases. But I'd expect we will be able to infer the happy path Architecture/Offer/Publisher/Sku out of a given nodepool.spec.release. This happy path use case where we don't want to put the burden in the API consumer to choose the happy path default image values is what I'm not seeing captured in the API

Reply:
let's move forward the PR. Captured this follow up here https://issues.redhat.com/browse/HOSTEDCP-1858